packages/firestore/src/core/query.ts

```diff
@@ -635,17 +610,14 @@ export class RelationFilter extends Filter {
   }
 }
 
+export class UnaryFilter extends FieldFilter {}
+
 /**
  * Filter that matches 'null' values.
  */
-export class NullFilter extends Filter {
-  constructor(public field: FieldPath) {
-    super();
-  }
-
-  matches(doc: Document): boolean {
-    const val = doc.field(this.field);
-    return val !== undefined && val.value() === null;
+export class NullFilter extends UnaryFilter {
```

Comment:
Concretely, why are we keeping NullFilter / NaNFilter rather than just relying on FieldFilter?  Removing them would be a nice simplification (we could probably get this PR to be net negative lines of code) and I'd rather forget a few resume tokens (in practice, NaN queries are probably nonexistent and null queries are probably rare-ish) than carry along these classes forever...

If we keep them, we should have some sort of comment explaining that they exist due to legacy canonicalIds  (unless there's another reason I'm missing)?